server/log_operation_manager.go

```diff
@@ -434,25 +451,35 @@ loop:
 		if err := l.getLogsAndExecutePass(ctx); err != nil {
 			glog.Errorf("failed to execute operation on logs: %v", err)
 		}
-
 		glog.V(1).Infof("Log operation manager pass complete")
 
-		// See if it's time to quit
-		select {
-		case <-ctx.Done():
-			glog.Infof("Log operation manager shutting down")
-			break loop
-		default:
-		}
-
 		// Wait for the configured time before going for another pass
 		duration := time.Since(start)
 		wait := l.info.RunInterval - duration
-		if wait > 0 {
-			glog.V(1).Infof("Processing started at %v for %v; wait %v before next run", start, duration, wait)
-			time.Sleep(wait)
-		} else {
-			glog.V(1).Infof("Processing started at %v for %v; start next run immediately", start, duration)
+		timeout := make(chan bool, 1)
+		go func() {
+			if wait > 0 {
+				glog.V(1).Infof("Processing started at %v for %v; wait %v before next run", start, duration, wait)
+				time.Sleep(wait)
+			} else {
+				glog.V(1).Infof("Processing started at %v for %v; start next run immediately", start, duration)
+			}
+			timeout <- true
+		}()
+
+		waiting := true
+		for waiting {
```

Comment:
Separately, I'm wondering if it might be clearer to crank the signing runs with a `time.Ticker`?

Something along the lines of this:

```golang
signNow := time.NewTicker(l.info.RunInterval)

for (true) {
select {
  case <-ctx.Done():
    signNow.Stop()
    // bye!
    break loop
  case r:= <-l.pendingResignations:
    r.execute(ctx)
  case <-signNow:
    // Processing started at %v ...
    if err := l.getLogsAndExecutePass(); {
      ...    
    }
}
```

Reply:
Good spot with the busy-spin; I reverted back to the previous structure and just added a new drain-resignations arm instead.

(Leaving the `Ticker` thing for another time.)